templates/config.json.j2

```diff
@@ -24,8 +24,28 @@
         "rpc": "{{ consul_rpc_bind_address }}",
       {% endif %}
     },
-    {% if consul_ports %}
-        "ports": {{ consul_ports | to_json }},
+    {## Ports Used ##}
+    {% if consul_ports | length > 0 %}
+      "ports": {{ consul_ports | to_json }},
+    {% else %}
+      "ports": {
+        {% if _consul_default_port_rpc | trim != '' %}
+          {% set _ = consul_ports.__setitem__('rpc', _consul_default_port_rpc) %}
```

Comment:
The template uses some jinja2 magic also present in the DebOps project to ensure that the consul_ports dict object is always identical to the config.json file values.

See https://stackoverflow.com/questions/36886650/how-to-add-a-new-entry-into-a-dictionary-object-while-using-jinja2
See https://stackoverflow.com/questions/4870346/can-a-jinja-variables-scope-extend-beyond-in-an-inner-block